subt/cloudsim2osgar.py

```diff
@@ -58,21 +60,45 @@ def publish(self, channel, data):
 
 
 class main:
-    def __init__(self):
+    def __init__(self, robot_name):
         # get cloudsim ready
-        robot_name = sys.argv[1]
-        imu_name = '/'+robot_name+'/imu/data'
         wait_for_master()
-        rospy.init_node('imu2osgar', log_level=rospy.DEBUG)
-        rospy.loginfo("waiting for {}".format(imu_name))
-        rospy.wait_for_message(imu_name, Imu)
-        rospy.sleep(2)
-        self.imu_count = 0
-
-        # start
+        rospy.init_node('cloudsim2osgar', log_level=rospy.DEBUG)
         self.bus = Bus()
-        self.bus.register('rot', 'acc', 'orientation')
-        rospy.Subscriber(imu_name, Imu, self.imu)
+
+        # common topics
+        topics = [
+            ('/' + robot_name + '/imu/data', Imu, self.imu, ('rot', 'acc', 'orientation')),
+        ]
+
+        # configuration specific topics
+        robot_description = rospy.get_param("/{}/robot_description".format(robot_name))
```

Comment:
are we sure, that we will get always the description? Or when `run_solution` has it then next call will succeed? Again I am not very sure if we want to have this logic (grep for keywords) on two places.

Reply:
> are we sure, that we will get always the description?

In the configurations we are currently supporting the description is available.

> Or when `run_solution` has it then next call will succeed?

That is also true - we don't get here if run_solution does not get it. Rosparams can be read as many times as necessary and I am not aware of anything that would remove the description from the ros param server.

> Again I am not very sure if we want to have this logic (grep for keywords) on two places.

There needs to be some kind of `if` anyway and this seemed like the easy way out (the only other way I could come up with involved creating some other strings representing configurations and synchronizing them between run_solution and cloudsim2osgar which seemed to me even worse).